### DISCUSSION TITLE
Not possible to use ipcRenderer?

### DISCUSSION THREAD

**Discussion by itsnwa** [cawa-93/vite-electron-builder#245] [Q&A] [answered]
I need to communicate with the main process, but `ipcRenderer.send` is throwing `__dirname is not defined`

```js
<script setup>
import { ipcRenderer } from 'electron';
const closeApp = () => ipcRenderer.send('closeApp');
</script>
```

**Comment by cawa-93** [comment, marked as answer]
https://github.com/cawa-93/vite-electron-builder#using-nodejs-api-in-renderer

**Reply by itsnwa** [reply]
Oh, great - thanks @cawa-93 :)

Found this contextBridge example in the electron docs quite helpful, if anyone else is wondering:
https://www.electronjs.org/docs/api/context-bridge#contextbridge

**Reply by cawa-93** [reply]
Another type-safe IPC example: https://github.com/cawa-93/vite-electron-builder/issues/189#issuecomment-824136546
